### DISCUSSION TITLE
Absolute paths in core.loadSkyImage() not working

### DISCUSSION THREAD

**Discussion by alphanimal** [Stellarium/stellarium#2012] [Q&A] [answered]
Hi!

When I use root.loadSkyImage() in a script, I cannot get it to load a file using an absolute path.

The [documentation ](https://stellarium.org/doc/0.21/classStelMainScriptAPI.html#a7f4d145e871a3b5f9b24a3ace11198d3) says:

> If a relative path is specified, "scripts/" will be prefixed before the image is searched for using StelFileMgr.

It seems that this is the case with absolute paths too. (I'm on Windows) For example:

`root.loadSkyImage('image1','/test.png' ...)` will try to load "%AppData%\Stellarium\scripts\test.png" and "%ProgramFiles%\Stellarium\scripts\test.png" - the same as using './test.png' or´just 'test.png'

'D:\test.png' will try to load '%AppData%\Stellarium\scripts\D:\test.png' etc.

Using / or \ doesn't make a difference.

Is there a way to load absolute paths or do I have to stick with the default paths?

Thanks!
Daniel

**Comment by gzotti** [comment]
I cannot remember the details here. Can you try file://D:/path/test.png? Or /D:/path/test.png?

**Comment by alphanimal** [comment]
They both don't work. procmon.exe tells me they landed at

- "C:\Users\Daniel\AppData\Roaming\Stellarium\scripts\D:\path\test.png"
- "C:\Program Files\Stellarium\scripts\D:\path\test.png"
- "C:\Users\Daniel\AppData\Roaming\Stellarium\scripts\file:\D:\path\test.png"
- "C:\Program Files\Stellarium\scripts\file:\D:\path\test.png"

**Comment by gzotti** [comment, marked as answer]
Indeed. At least since 2009, "scripts/" is prepended to any given filename. There is no way to specify an absolute path directly. I have fixed the documentation accordingly.

**Reply by alphanimal** [reply]
Alright I'll just use the scripts folder, thanks!

**Comment by alphanimal** [comment]
fyi in the [documentation for core.loadSkyImage](https://stellarium.org/doc/0.21/classStelMainScriptAPI.html#a7f4d145e871a3b5f9b24a3ace11198d3) it says:

> Images in AzAlt frame are not affected by atmosphere effects like refraction or extinction.

I think that's wrong. The image definitely seems to be affected by refraction. Extinction I can't tell from just looking at it.
(when I switch on/off the atmosphere I can see the image getting squished near the horizon)

**Reply by gzotti** [reply]
OK, I must have a look. Maybe this was changed meanwhile.
